azuki-core/src/main/kotlin/com/anaplan/engineering/azuki/core/runner/JUnitScenarioRunner.kt

```diff
@@ -230,6 +237,20 @@ class JUnitScenarioRunner<
 
     class SkippedException(msg: String) : Exception(msg)
 
+    private val parameterMethod: FrameworkMethod? by lazy {
+        val companionObject = kClass.companionObjectInstance
+        if (companionObject == null) {
+            null
+        } else {
+            val parameterMethods = TestClass(companionObject.javaClass).getAnnotatedMethods(Parameters::class.java)
+            Log.debug("Parameter methods: $parameterMethods")
+            if (parameterMethods.size > 1) {
+                throw IllegalStateException("Multiple parameter methods declared for $testClass")
+            }
+            if (parameterMethods.isEmpty()) null else parameterMethods.first()
```

Comment:
⛏️ To me, this would have been more clear as
```suggestion
            when (parameters.size) {
              0 -> null,
              1 -> parametersMethods.first(),
              else -> throw IllegalStateException("Multiple parameter methods declared for $testClass")
            }
```